lib/ansible/modules/system/ufw.py

```diff
@@ -296,20 +297,36 @@ def ufw_version():
     (_, pre_state, _) = module.run_command(ufw_bin + ' status verbose')
     (_, pre_rules, _) = module.run_command("grep '^### tuple' /lib/ufw/user.rules /lib/ufw/user6.rules /etc/ufw/user.rules /etc/ufw/user6.rules")
 
+    changed = False
     # Execute commands
     for (command, value) in commands.items():
         cmd = [[ufw_bin], [module.check_mode, '--dry-run']]
 
         if command == 'state':
             states = {'enabled': 'enable', 'disabled': 'disable',
                       'reloaded': 'reload', 'reset': 'reset'}
-            execute(cmd + [['-f'], [states[value]]])
+            if module.check_mode:
+                if value in ['reloaded', 'reset']:
+                    changed = True
+                else:
+                    (_, ufw_state, _) = execute([[ufw_bin], ["status"]])
+                    ufw_enabled = ufw_state.find("active") != -1
+                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and ufw_enabled):
```

Comment:
There's a `not` missing somewhere in here :)
```suggestion
                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and not ufw_enabled):
```